### ISSUE TITLE
Decorators method for blocking timelimiter

### ISSUE THREAD

**Issue by SimonScholz** [resilience4j/resilience4j#1575]
Resilience4j version: 1.7.2

Java version: 11

Do you have any suggestion to easily come up with a fallback for a "BLOCKING" timelimiter?

![image](https://user-images.githubusercontent.com/7559962/139912439-f7ec642b-5719-44e6-bfda-884610603b4c.png)

The io.github.resilience4j.decorators.Decorators.DecorateCompletionStage#withTimeLimiter method needs a `ScheduledExecutorService`, which I do not want to provide in my blocking Spring Boot WebMVC application.

Basically I'd like to have a 
```
T blockingResult = Decorators.ofSupplier(supplier)
    .withTimeLimiter(timeLimiter)
    .withFallback(asList(TimeoutException.class),  
                  throwable -> "Hello from Recovery").get()
```

Or is there any static helper method to wrap a regular `TimeLimiter` inside a fallback decorator?
Of course I could wrap this in a try catch block, but having such a `withFallback` method would be way more elegant.

**Comment by RobWin** [comment]
Hi,
Resilience4j can't limit the execution time of a `Supplier`. 
The `Decorators` class does not support `Supplier<Future<T>>` yet. But you can make use of our `CallableUtils`

```
Callable<String> decorated = timeLimiter.decorateFutureSupplier(futureSupplier);
decorated = CallableUtils.recover(decorated, TimeoutException.class, (ex) -> "Hello from Recovery");
```
        assertThat(decorated.call()).isEqualTo("Hello from Recovery");

**Comment by SimonScholz** [comment]
Thanks a lot for your input.
Closing the issue for now, but feel free to re open it in case you'd want to adjust the `Decorators` class for this purpose :-)
